lang/cs/Org.Apache.REEF.IMRU.Tests/TestSystemStates.cs

```diff
@@ -0,0 +1,135 @@
+        /// Test Equal method for StateTransition
+        /// </summary>
+        [Fact]
+        public void TestStateTransitionEquals()
+        {
+            var s1 = new StateTransition<SystemState, SystemStateEvent>(SystemState.WaitingForEvaluator,
+                SystemStateEvent.AllContextsAreReady);
+            var s2 = new StateTransition<SystemState, SystemStateEvent>(SystemState.WaitingForEvaluator,
+                SystemStateEvent.AllContextsAreReady);
+            Assert.True(s1.Equals(s2), "The two transitions should be equal");
+        }
+
+        /// <summary>
+        /// Successful transitions from WaitingForEvaluator to TasksCompleted
+        /// </summary>
+        [Fact]
+        public void TestFromRequestEvaluatorToTasksComplete()
+        {
+            var systemState = new SystemStateMachine();
+            Assert.True(systemState.CurrentState.Equals(SystemState.WaitingForEvaluator), "The initial state should be WaitingForEvaluator.");
+            Assert.True(systemState.MoveNext(SystemStateEvent.AllContextsAreReady).Equals(SystemState.SubmittingTasks), "Fail to move from WaitingForEvaluator state to SubmittingTasks state.");
+            Assert.True(systemState.MoveNext(SystemStateEvent.AllTasksAreRunning).Equals(SystemState.TasksRunning), "Fail to move from SubmittingTasks state to TasksRunning state.");
+            Assert.True(systemState.MoveNext(SystemStateEvent.AllTasksAreCompleted).Equals(SystemState.TasksCompleted), "Fail to move from TasksRunning state to TasksCompleted state.");
+        }
+
+        [Fact]
+        public void TestNoRecoverableFailedEvaluatorDuringWaitingForEvaluator()
+        {
+            var systemState = new SystemStateMachine();
+            Assert.True(systemState.MoveNext(SystemStateEvent.NotRecoverable).Equals(SystemState.Fail), "Fail to move from WaitingForEvaluator state to Fail state.");
+        }
+
+        /// <summary>
+        /// Recovery scenario, from WaitingForEvaluator->SubmittingTasks->TasksRunning->ShuttingDown->WaitingForEvaluator
+        /// </summary>
+        [Fact]
+        public void TestFailedEvaluatorAfterTasksAreRunning()
+        {
+            var systemState = new SystemStateMachine();
+            systemState.MoveNext(SystemStateEvent.AllContextsAreReady);
+            systemState.MoveNext(SystemStateEvent.AllTasksAreRunning);
+            Assert.True(systemState.MoveNext(SystemStateEvent.FailedNode).Equals(SystemState.ShuttingDown), "Fail to move from TasksRunning state to ShuttingDown state.");
+            Assert.True(systemState.MoveNext(SystemStateEvent.Recover).Equals(SystemState.WaitingForEvaluator), "Fail to move from ShuttingDown state to WaitingForEvaluator state.");
+        }
+
+        /// <summary>
+        /// Recovery scenario, from WaitingForEvaluator->SubmittingTasks->TasksRunning->ShuttingDown->WaitingForEvaluator
+        /// </summary>
+        [Fact]
+        public void TestFailedTaskAfterTasksAreRunning()
```

Comment:
Since we're not distinguishing failed evaluator and failed task any more, `TestFailedTaskAfterTasksAreRunning` test becomes identical to `TestFailedEvaluatorAfterTasksAreRunning`. 

Same, `TestFailedEvaluatorDuringTaskSubmittingThenFail` = `TestFailedTaskDuringTaskSubmittingThenFailToRecovery`, and `TestFailedEvaluatorDuringTaskSubmittingThenRecover` = `TestFailedTaskDuringTaskSubmittingThenFailToRecovery`